### ISSUE TITLE
Vulkan Device Loss - that weren't occuring before - happening now during random situations in lastest Yuzu EAs 

### ISSUE THREAD

**Issue by Flexa007** [yuzu-emu/yuzu#10600]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Early Access 3602 onwards (last worked perfectly EA 3601)

### Description of Issue

A Vulkan "device loss" started happening from Yuzu EA 3602 onwards (currently 3635, as of this post date and time) and has remained ever since, all versions, none excluded - tried all.

This seems very puzzling to me, as when switch back to run again in 3601, I'm able to maintain stable running for hours, with no single crash or let alone even freezing, keeping a variation of 42-44 to 51-53 and occasional dips to 33-37 in most games (TLZ: TOTK, especially).

To confirm, I've tried and tested over the course of this week under many circunstances in 3601, TOTK case:

1) Let the game nearly an entire day afk in "Lookout Landing" (one of highest stress areas), just left the game/Yuzu on the open there, not in pause, nor in menu or map, and was still able to play for many hours without any single issue after with no PC reboot nor restarting Yuzu. The next day, tried again, this time leaving it open in "Kakariko Village" (another high stress area). Same results. No problem whatsoever playing as normally picking up right after, even with PC and Yuzu on and emulating for nearly 24 hours.

2) Tested most different setups between all graphical/emulation settings, be it 2x, 3x, accuracy level, BC1, BC3 or  Uninc texs, Fast GPU, the various asynchronous options and CPU safe/unsafe, etc... All without issue in 3601. 

Now, from EA 3602 (all the way up to current 3635) this Vulkan device loss keeps happening in situations cannot figure exactly why. It occurs in seemingly very random moments. Strangely enough, it doesn't happen in the high stress areas or when the game is under some "heavy" load, so for some reason, can play normally and even spam actions and abilities inside areas like Fire Temple or Lookout Landing or Kakariko, for example, but then the error happens in some seemingly softer overworld area, out of nowhere, and this makes it all the more puzzling to understand. I was initially obviously suspecting some RAM usage, but using OpenGL demonstrates even a higher RAM load and it simply doesn't crash. Nonetheless, as pointed, using even heavier settings returning back to 3601 this doesn't happen. Even if putting a special profile in Nvidia driver with everything set to maximum, no crashes in Vulkan/uninc texs in EA 3601.

But then, from 3602 onwards, even trying BC1, 1x, it happens.

I'm especially sorry if bothering you, guys. I've tried everything over the course of this week and couldn't come up with a solution, so this is a last resort. I don't know how to exactly reproduce this device loss error, because it happens in most random and seemingly uncalled times from EA 3602-3635, as said, without the RAM being in the limit (or even CPU or GPU limit for the record) and for some reason, as previously also told, this not being the case even putting everything to max in 3601 in both Yuzu and Nvidia 3d settings itself. 

I wonder if any change made from 3601 to 3602 ( and that has remained since then), caused some sort of hard incompatibility with systems similar to mine. Hopefully, this might help some people in similar curious situation, as well.

To help understanding, brought the log of one such error (EA 3635).

### Expected Behavior

No device loss, same as 3601.

### Reproduction Steps

As reported I'm not exactly sure how to reproduce the error, since it happens at complete random but not during the heavier areas like Kakariko or Fire Temple, when the RAM usage is even higher.  And for some reason, as previously also told, this not being the case even putting everything to max in 3601 in both Yuzu and Nvidia 3d settings itself. 

Log following to help clear if any issue. 

### Log File

[yuzu_log.txt.old.txt](https://github.com/yuzu-emu/yuzu/files/11644472/yuzu_log.txt.old.txt)

### System Configuration

CPU: I7 7700k
GPU/Driver: 1080ti - 535.98 
RAM: 16GB
OS: W11 (tested in both W10 and W11. Same results, as well)

**Comment by liamwhite** [comment]
If you can provide a log with graphics debugging enabled, this would be the most helpful in getting a proper resolution.

**Comment by Flexa007** [comment]
> If you can provide a log with graphics debugging enabled, this would be the most helpful in getting a proper resolution.

Thanks for the reply, Liam. 

Sorry for taking long, I was recording and getting graphic logs in both versions to provide better info.  

Took a gameplay of around 10 minutes in both 3635 and 3601. Both have absolute same settings in everything.

At the end of 3635, I could indirectly reproduce it as said error happened.

**3635 log**:
[3635 graphic yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11644754/3635.graphic.yuzu_log.txt)

The exact same session for the log above was recorded. It also provides proof of what said initially. As am able to notice, can mantain a smooth FPS around 50ish+ and the error occurs in a moment of no such stress, while frames are still stable and high, for example. Not even visibly freezing or lagging.

3635 recorded session (from log above):
 [https://www.youtube.com/watch?v=JzhkFBSuL8E&ab_channel=RodrigoFigueiredo](url)

Now following with 3601 files from the same area, as can see. In this one, as also initially said, no error whatsoever. Again mantaining similar smooth frame rate. I stopped yuzu at the end as this is stable (therefore, could go for hours if necessary) and already got around the time mark of the previous attempt. Providing both EA logs with similar gameplay/emulation time, for any comparisons.

**3601 log**:
[3601 graphic yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11644772/3601.graphic.yuzu_log.txt)

3601 recorded session (from log above):
[https://www.youtube.com/watch?v=gFOJDjKnqek&ab_channel=RodrigoFigueiredo](url)

Hopefully, this might help with this situation and thanks again

**Comment by liamwhite** [comment]
Sorry, I forgot to mention, you need to have the Vulkan Validation Layer installed for the graphics debugging output to produce anything useful. Also, I only need the log for a build that gets device losses.

**Comment by Flexa007** [comment]
> Sorry, I forgot to mention, you need to have the Vulkan Validation Layer installed for the graphics debugging output to produce anything useful. Also, I only need the log for a build that gets device losses.

Much appreciated for the response, Liam.

I've followed your direction, and while noticed a performance hit of around 40-45%ish while running, was able to reproduce apparently the same "device loss 627". Hopefully, it has the same origin in this case (and not due the occasional heavy loading). All settings and configs remained the same as previous.

Please, if possible, confirm if is this the format you've requested:

**EA 3635 graphic log (Vulkan validation layers installed)**
[3635 Vulkan validation layers yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11644880/3635.Vulkan.validation.layers.yuzu_log.txt)

**Comment by liamwhite** [comment]
This is the correct format, unfortunately the validation layers didn't show any invalid API usage before the device hang/fault occurred, so it's hard to say what the problem is.

**Comment by liamwhite** [comment]
I looked through the release history and 3602's only change adds this PR: https://github.com/yuzu-emu/yuzu/pull/10402
Unclear how this would cause device losses though.

**Comment by Flexa007** [comment]
> I looked through the release history and 3602's only change adds this PR: #10402 Unclear how this would cause device losses though.

That's quite intriguing. Actually, looking thru the matter and now that you mention PR: #10402 and that being the change between these versions, I got something to report in this regard. Maybe it might help in a way.

Even using EA 3601, in theory before the fix, I don't have the pixelated ultrahand. I suspect the lastest Nvidia driver (535.98), launched very recently, might have incidentally/colleterally touched upon it natively while progressing/troubleshooting Vulkan, perhaps. 

I took a demonstration in 3601, It renders perfectly fine against uneven surfaces and solid/straight ones. No pixelization be it horizontally or vertically, fast or slow movement. 

https://www.youtube.com/watch?v=sYvumB-83xM&ab_channel=RodrigoFigueiredo

Curiously enough, for GeForce users expressing compatibility issues with some of the lastest builds and having to maintain previous versions, could be interesting for them to notice if pixelization still persists when using lastest Nvidia driver, as well.

**Comment by liamwhite** [comment]
> in theory before the fix, I don't have the pixelated ultrahand

It's a combination of GPU and driver, and is arguably a bug in the game, not in the emulator.

There are two meshes projected onto the surrounding surfaces which have the same depth values as well as alpha in the pass, but there isn't an explicit barrier between the draw for each. On hardware, this ends up working out, because the amount of parallel fragment invocations is low enough for them to never interfere with each other. On desktop, nondeterminism starts occurring for systems with intermediate-to-high numbers of shaders, because the shader invocations start interfering with each other due to driver scheduling.

When the shader count is extremely high, and all sample invocations are dispatched in parallel, you might never see the issue at all because they'll act as if there was a copy of the framebuffer between them. You can see #10560 for a more pernicious case of this.

I never experienced the ultrahand projection issue with AMDVLK or RADV on my 6700XT, but I got it consistently with AMDVLK on the RX 560 4GB, and with the 3060Ti on the latest Windows drivers.

**Comment by liamwhite** [comment]
[yuzu.zip](https://github.com/yuzu-emu/yuzu/files/11645992/yuzu.zip)
Here's a Windows version of yuzu built against master that has the extra renderpass breaks removed This gives me the ultrahand issue again. Does this resolve the device losses?

**Comment by Flexa007** [comment]
> > in theory before the fix, I don't have the pixelated ultrahand
> 
> It's a combination of GPU and driver, and is arguably a bug in the game, not in the emulator.
> 
> There are two meshes projected onto the surrounding surfaces which have the same depth values as well as alpha in the pass, but there isn't an explicit barrier between the draw for each. On hardware, this ends up working out, because the amount of parallel fragment invocations is low enough for them to never interfere with each other. On desktop, nondeterminism starts occurring for systems with intermediate-to-high numbers of shaders, because the shader invocations start interfering with each other due to driver scheduling.
> 
> When the shader count is extremely high, and all sample invocations are dispatched in parallel, you might never see the issue at all because they'll act as if there was a copy of the framebuffer between them. You can see #10560 for a more pernicious case of this.
> 
> I never experienced the ultrahand projection issue with AMDVLK or RADV on my 6700XT, but I got it consistently with AMDVLK on the RX 560 4GB, and with the 3060Ti on the latest Windows drivers.



> [yuzu.zip](https://github.com/yuzu-emu/yuzu/files/11645992/yuzu.zip) Here's a Windows version of yuzu built against master that has the extra renderpass breaks removed. This gives me the ultrahand issue again.
> 
> ```diff
> --- a/src/video_core/renderer_vulkan/vk_texture_cache.cpp
> +++ b/src/video_core/renderer_vulkan/vk_texture_cache.cpp
> @@ -862,7 +862,7 @@ VkBuffer TextureCacheRuntime::GetTemporaryBuffer(size_t needed_size) {
>  }
> 
>  void TextureCacheRuntime::BarrierFeedbackLoop() {
> -    scheduler.RequestOutsideRenderPassOperationContext();
> +    //scheduler.RequestOutsideRenderPassOperationContext();
>  }
> 
>  void TextureCacheRuntime::ReinterpretImage(Image& dst, Image& src,
> ```
> 
> Does this resolve the device losses?

Liam, first, thanks for the support and for explaining and sheding light at the issue thoroughly. Really helped comprehend the matter at hand to the best.

Secondly, sorry again to take long to answer, but wanted to provide directly a substantial response from the the get go. 

And so far, conducted a 3 hour run with the build you've provided, this concerned: 

- around 30 minutes in Overworld in blizzard conditions (areas around Rito Village),
- around 30 minutes in Great Sky Islands,
- around 40 mins in the Depths overall, 
- around 20 minutes inside Fire Temple, 
- around 10 minutes running and activating/deactivating FOV in Kakariko, 5 minutes spamming ultrahand. 
- around 10 minutes running and activating/deactivating FOV in Lookout Landing, 5 minutes spamming ultrahand.
- around 30 mins in same Eldin area as the previous logs/tests,

During the course of this time, **I can report using the modified build you've made, not a single device loss happened. It ran perfectly.**

The only time I've experienced something in this regard was a slight freeze (FPS drop to 7) while in the blizzard area close to Rito, but it immediately recovered back to normal, so I believe it wasn't related. 

Then testing EA 3636, device loss happened with approximately 10 mins, as similarly demonstrated before. 

Both tests, build master-125a0e5a0 and EA 3636 were done using the exact same settings as the ones before, both in yuzu and Nvidia driver.

I can also report that using builld master-125a0e5a0, I continue to not have pixelated ultrahand nor signs of it, even without the code. To provide visual description, at the end of the 3 hour test run I recorded it in different surfaces for demonstration:

https://www.youtube.com/watch?v=JWaILpVcwsE&ab_channel=RodrigoFigueiredo

Can also report that the build didn't break anything else. No other common glitches were observed, this includes;

- Haven't seen render/tex cuts in the Depths; 
- Haven't experienced TP light corruption for the most part (as seen in video above itself, even against a very bright background). Only see one small TP effect breaking against thunderstorm background;
- No gloom textures glitches (this one I never experienced, personally. But have to include it didn't happen with this build, as well).

**Additionally, as already pointed, all tests happened under same conditions and seetings. In this case, this means the 3 hour run did go with Uninc texs and, again, went smoothly, perfectly fine, without a single occurance of device loss.**

 The log of the run (just the basic one, this confirms the build and the same settings as the previous attempts), and as can see, stopping emulation plainly and smoothly by the end:

[yuzu - builld master-125a0e5a0 - 3 hour test run - basic log.txt](https://github.com/yuzu-emu/yuzu/files/11646456/yuzu.-.builld.master-125a0e5a0.-.3.hour.test.run.-.basic.log.txt)

I can only thank for the work, for the time you took for this matter and for the support, Liam. This is absolutely stunning. Extremely promising.

If need be, I can also conduct another runs, providing most specific/detailed logs or attempting different tests.

**Comment by liamwhite** [comment]
I've pushed a branch [here](https://github.com/liamwhite/yuzu/commit/903fd393e3f422d38546e327fd2142a5f8711eb5) that doesn't end the renderpass before inserting the barrier. Maybe this is causing the issue.
Windows build: [yuzu.zip](https://github.com/yuzu-emu/yuzu/files/11646594/yuzu.zip)

**Comment by Flexa007** [comment]
> I've pushed a branch [here](https://github.com/liamwhite/yuzu/commit/903fd393e3f422d38546e327fd2142a5f8711eb5) that doesn't end the renderpass before inserting the barrier. Maybe this is causing the issue. Windows build: [yuzu.zip](https://github.com/yuzu-emu/yuzu/files/11646594/yuzu.zip)

Liam, reporting the results:

- method: only pasted the respective exe into EA 3637 folder. No other files modified. Settings equal to all previous tests.

Results:

- build 2 (deemed **"subpass barrier 903fd393e"**): device loss occured with approximately 7-8 minutes, while in Lookout Landing.
- build 1 (deemed **"master-125a0e5a0"**): currently running 1 hour, and so far again without any single device loss/crash happening, in such, up to now repeating the stability of previous run. Immediately tested inside Lookout Landing extensively.

**Comment by liamwhite** [comment]
So it's the barrier then, not the renderpass. I find it increasingly likely this is an nvidia driver bug given that there is no evidence of validation errors preceding the call.

**Comment by Squall-Leonhart** [comment]
I'm on the same drivers with a 1060, i don't have these device losses, i do have the ultra hand glitches.

**Comment by Flexa007** [comment]
> So it's the barrier then, not the renderpass. I find it increasingly likely this is an nvidia driver bug given that there is no evidence of validation errors preceding the call.



> I'm on the same drivers with a 1060, i don't have these device losses, i do have the ultra hand glitches.
> 
> ![image](https://user-images.githubusercontent.com/1098176/243224252-a6cf6cb0-8938-457b-96dc-55fdb71e0873.png)
> 
> Op, do you have Hardware Accelerated Graphics Scheduling turned on?
> 
> and can you post [vkdiag ](https://github.com/13xforever/vkdiag) output

Thanks for the support, all. I wasn't able to answer earlier.

I forgot to add this before, but from the "master-125a0e5a0" build, regarding common glitches, in addition to the ones mentioned that I haven't experienced, I also am not having the "recall circle" one. Was running trying different setups and so far no device loss or any graphical glitches. 

This one even in normal acc and without Fast GPU enabled:

![Settings](https://github.com/yuzu-emu/yuzu/assets/134863891/0dff263f-3fbf-4818-a135-50674df12a0b)
![TOTKExhibit1Recall](https://github.com/yuzu-emu/yuzu/assets/134863891/a9141e1f-2383-4215-b0b0-0cd5c5847fe3)

Trying one after another, repeatedly:
https://www.youtube.com/watch?v=0k7rPt_ZM4o&ab_channel=RodrigoFigueiredo

It remains same with it enabled. So far, no visible glitches. Gotta add haven't experienced invisible gloom damage in Depths, as well, even running these same settings (Normal Acc, unchecked Fast GPU).

@Squall-Leonhart Thanks, I'll proceed with the check right away. 

And yep, enabled by default. I haven't modified anything in current Windows yet. Not even any simplier changes, nor any registry stuff. It is a fresh install from the course of the personal tests during this past week. Coincidentally, the day I've installed it was exactly the same day 535.98 released, both in May 30. I straight downloaded it (Nvidia's 535.98) into a fresh W11 as first ever driver, even before Windows Update could pick their default version (installed Windows without network). 

**Edition** Windows 11 Pro 
**Version** 22H2
**Installed on** **_05/30/2023_**
**OS Build** 22621.1702
**Experience** Windows Feature Experience Pack 1000.22641.1000.0 

Gfx driver:

Release 535 Driver for Windows, 
Version 535.98
**_30 May, 2023._** 

The release notes do mention some Vulkan notes, but not sure if this might be interesting. Nonetheless:
https://us.download.nvidia.com/Windows/535.98/535.98-win11-win10-release-notes.pdf

Curiously, the first build provided by @liamwhite is making the game run perfectly by all accounts. Even in terms of fps, it is very smooth. This machine is outdated, so I'm quite surprised by this performance (thanks yuzu team). All the videos I've uploaded here also serve to visually confirm and also express what I mean by this. Excellent stuff.

**Comment by Squall-Leonhart** [comment]
HAGS could be both the source of the TDR's and the cause of the ultrahand fix, as it shifts a bunch of scheduling back towards the GPU, it basically changes up the timings of a lot of behaviors of the graphics driver which would usually be performed on a high priority thread.

**Comment by liamwhite** [comment]
I am working on a potential barrierless solution.

**Comment by Flexa007** [comment]
@Squall-Leonhart tested with W11 hardware acceleration disabled, still same results in both builds. 2 hours running with no issues whatsoever in "master-125a0e5a0", while a random device loss happened within around 5-6 mins in Overworld, Hyrule Field area during "subpass barrier 903fd39", and a device loss happened while in Faron stable area for the build "EA 3638", with approximately 3 minutes a brief after fast traveling. 

Still regarding HAGS setting, I wasn't able to notice any visible performance difference with it disabled.

And also under these said circunstances, still no pixelated ultrahand:

https://www.youtube.com/watch?v=loST2S-ASkk&ab_channel=RodrigoFigueiredo

Additionally, I repeatedly attempted running the VKDiag script, it starts the initial check but then closes as I press Enter for the 32/64 part, even with lastest Power Shell (7.3) and trying under unrestriced/bypass, and "unblocking" all the files manually. If you could kindly provide an alternative way or manner, I would be mostly glad to return data.


@liamwhite I can now confirm it wasn't any kind of fluke or some great amount of luck regarding build "master-125a0e5a0", I ran it actively for about 2 hours yesterday, left afk mid Kakariko overnight (in open view. Not in menu nor map), played approx 15 minutes during morning before leaving again, returned from office and been playing - as I type this - for about 3,5 hours, for a total around of 19 hours non stop yuzu emulation, nor PC reboot. No single device loss.

**Comment by Squall-Leonhart** [comment]
Video is marked private.

> Additionally, I repeatedly attempted running the VKDiag script, it starts the initial check but then closes as I press Enter for the 32/64 part, even with lastest Power Shell (7.3) and trying under unrestriced/bypass, and "unblocking" all the files manually. If you could kindly provide an alternative way or manner, I would be mostly glad to return data.

I'm not seeing anything like that in the output from the Exe in the releases, there could be some weird vulkan layer installed that is crashing the vulkan loader.

it should appear as

```[+] VkDiag version: 1.2.0
[i] CPU: Intel(R) Xeon(R) CPU E5-1680 v2 @ 3.00GHz
[+] OS: Microsoft Windows 11 Pro
[+] Version: 10.0.22621 (Windows 11 22H2)
[+] System Vulkan loader version: 1.3.241.0

Found 1 active GPU:
[v] NVIDIA GeForce GTX 1060 6GB
    [+] Driver version: 31.0.15.3598 (5-24-2023)
    [v] Proper Vulkan driver registration

Vulkan registration information:
[+] No explicit driver registration entries
[+] Implicit layers registration (HKEY_LOCAL_MACHINE, 64-bit):
    [+] Debugging capture layer for RenderDoc, v1.22.0.0, API v1.3.131 (renderdoc.json)
    [+] RTSS overlay hook bootstrap, API v1.3.224 (RTSSVkLayer64.json)
    [+] Steam Overlay, API v1.3.207 (SteamOverlayVulkanLayer64.json)
    [+] Steam Pipeline Caching, API v1.3.207 (SteamFossilizeVulkanLayer64.json)
[+] Implicit layers registration (HKEY_LOCAL_MACHINE, 32-bit):
    [+] Debugging capture layer for RenderDoc, v1.22.0.0, API v1.3.131 (renderdoc.json)
    [+] RTSS overlay hook bootstrap, API v1.3.224 (RTSSVkLayer32.json)
    [+] Steam Overlay, API v1.3.207 (SteamOverlayVulkanLayer.json)
    [+] Steam Pipeline Caching, API v1.3.207 (SteamFossilizeVulkanLayer.json)
[+] Explicit layers registration (HKEY_LOCAL_MACHINE, 64-bit):
    [+] Rockstar Games Social Club, v2.1.6.7, API v1.0.70 (SocialClubVulkanLayer.json)
[+] Explicit layers registration (HKEY_LOCAL_MACHINE, 32-bit):
    [+] Rockstar Games Social Club, v2.1.6.7, API v1.0.70 (SocialClubVulkanLayer.json)
Press any key to exit the tool...is screen content for support.```

**Comment by Flexa007** [comment]
@Squall-Leonhart done and done! Sorry about the videos. My bad. I believe they are all accordingly now.

The VkDiag display:

![Sem título](https://github.com/yuzu-emu/yuzu/assets/134863891/1f730045-c2e6-4c27-a48a-6f4592f7f4cc)

**Comment by Flexa007** [comment]
@liamwhite noticed it quite incidentally while checking something else out of curiosity, but gotta report the build also does seem to provide/behave, for some reason, for some impressive RAM management on my end, as well. With what seems to demonstrate apparently no signs of any leak. This was during the lastest lengthy run, LL:

![LL ram usage end](https://github.com/yuzu-emu/yuzu/assets/134863891/9de1daf3-abab-408e-ae08-371db5276e21)
With gfx settings as:
![LL ram gfx settings1](https://github.com/yuzu-emu/yuzu/assets/134863891/c9e61c1f-6cc2-4a4a-96af-d7364cea0027)
![LL ram gfx settings2](https://github.com/yuzu-emu/yuzu/assets/134863891/b33fc3a7-c52a-4583-95cd-4439b03bff77)

Vid of the moment till taking pictures above:

https://www.youtube.com/watch?v=biRYa8rdPQY&ab_channel=RodrigoFigueiredo

Making a recheck later on:

![LL ram](https://github.com/yuzu-emu/yuzu/assets/134863891/e8b16813-565b-4381-8413-358ee6029fee)

Settings remained same:

![LL gfx2](https://github.com/yuzu-emu/yuzu/assets/134863891/c2f194c3-b96e-4943-8e19-514ecc563589)
![LL gfx3](https://github.com/yuzu-emu/yuzu/assets/134863891/dbe588b4-efce-4302-8157-433b0d51622c)

Although Task Manager is not exactly the best source, this is still a very curious display of RAM usage. Seems optimal enough.

P.S: did now a 30 minute observation from the getting go (launching yuzu, then immediatelly launching the game) as soon as loaded straight in LL, RAM displayed 5,8-6ish then started slowly dropping back. By the end, was already 2,6-5ish.

Same gxf settings as above. Uncomp texs.

**Comment by liamwhite** [comment]
Barrierless attempt didn't work out. I think my only option is to try to implement fragment shader interlock for the feedback loops. No ETA on that.

**Comment by Squall-Leonhart** [comment]
I believe both the lack of the issue that required the barrier in the first place, and the crashes with the barrier might be driver misbehaver specific to certain GPU's in the current driver.

**Comment by Flexa007** [comment]
@liamwhite @Squall-Leonhart No worries, take your time. I'm happy the issue was able to be studied and identified and hopefully a future workaround/fix may help people who might be in same situation, especially those who don't even know what might be happening. Thanks a lot and good work all around! If need anything I could provide, please ask anytime.

**Comment by liamwhite** [comment]
I ordered a 1080Ti to see if I can get this to happen myself and will report back after testing. (No big deal if I can't, needed a desktop Pascal anyway.)

**Comment by LOuroboros** [comment]
The same thing seems to be happening on my end with Persona 5 Royal.

I was wondering what was going on since I've been playing that game using Yuzu for over 120 hours with not a single crash, and now, after updating to the latest mainline I got 2 crashes in the past few hours of playtime.

> [3983.563229] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:578: 0xadd92bf55ce796de
> [3983.563282] Shader <Warning> shader_recompiler\frontend\maxwell\translate\impl\move_special_register.cpp:Read:141: (STUBBED) SR_WSCALEFACTOR_XY
> [3983.563285] Shader <Warning> shader_recompiler\frontend\maxwell\translate\impl\move_special_register.cpp:Read:144: (STUBBED) SR_WSCALEFACTOR_Z
> [3983.563320] Shader <Warning> shader_recompiler\frontend\maxwell\translate\impl\vote.cpp:VOTE_vtg:50: (STUBBED) called
> [3983.563323] Shader <Warning> shader_recompiler\frontend\ir\ir_emitter.cpp:GetFlowTest:264: (STUBBED) FCSM_TR
> [4004.807523] Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:627: Device loss occurred!

Since my hardware's lack of power doesn't seem to be directly related to the problem at hand *(Yuzu doesn't even make full use of my GPU)*, I'd like to leave a log here just in case it can be useful at all.

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11699995/yuzu_log.txt)

As the log says, and since this seems to be a GPU related problem primarily, I have an NVIDIA GeForce GT 1030 *(more specifically the model made by Gigabyte)*, Driver Version 512.95.

Funnily enough, the reason why I'm using such an old version was exactly to work around this same sort of "Vulkan Device Loss" error, [which some people including me were getting on Pokémon: Legends Arceus a while back when using the latest driver available](https://github.com/yuzu-emu/yuzu/issues/9591#issuecomment-1489516419).

**Comment by Flexa007** [comment]
@LOuroboros Thanks for the report and what a relief it also worked for you (and the very same build). The data you provided I take it will contribute to better comprehend the matter at hand.

@liamwhite Oh, wow. Thank you for the effort! Hopefully, it'll allow you to be able to perfectly study the situation.

In a suggestion, and first of all, I'm incredibly sorry if is this completely basic ignorance over my layman (lack of) knowledge about details of programming, however, would it be in the realm of possibility an optional setting, perhaps in "Advanced Graphics Settings" window to check or uncheck the render barrier/situation? With the choice as option, might be a good compromise between those who precisely need it and run smoothly, and those who might be experiencing contrivances during playing time. Then again, please, greatly pardon me if this is not possible from the  program side.

**Comment by Squall-Leonhart** [comment]
The intent is to emulate the behavior of the switch graphics pipeline as close to perfect as possible, this introducing a crash is a bit of an unusual thing given it is being difficult to replicate on the same but earlier iteration o the arrchitecture.

It does make a bit of sense for the 1030 and 1050 to have similar issues to the 1080ti as some changes that deviate from the 104/106 gpu's are in the 102, 107 and 108 chips.

**Comment by JJ1553** [comment]
Hi, I was just sent over here VIA reddit due. As it turns out, I have the same issue with a relatively large FPS drop with the upgrade to version EA 3602 from 3601, roughly a 25% drop from 28 to 21 fps in lookout landing. It does seem I am running different hardware than Flexa with a Ryzen 5 3600 and Radeon 6700xt (both overclocked). If you guys would like any more information from me for debugging purposes I would be happy to help!

**Comment by Squall-Leonhart** [comment]
that cpu and gpu combo should manage solid 30, your settings are too high somewhere.

**Comment by Flexa007** [comment]
> Hi, I was just sent over here VIA reddit due. As it turns out, I have the same issue with a relatively large FPS drop with the upgrade to version EA 3602 from 3601, roughly a 25% drop from 28 to 21 fps in lookout landing. It does seem I am running different hardware than Flexa with a Ryzen 5 3600 and Radeon 6700xt (both overclocked). If you guys would like any more information from me for debugging purposes I would be happy to help!

Hello, JJ

I believe some comparisons would be most welcoming for the devs to have a clearer or more concrete idea and data over how these specific 3602 changes may been visibly taxing your system in regards to previous versions.

Still on this matter, there's also a couple of builds provided by @liamwhite in this very same thread, as test compilations for pinpointing such issue, that could be of use for you to try, as well.  Especially the first one (removes entirely aforementioned changes), so far, has been proven fruitful collecting data to understand the situation. You might want check and report your own findings about it, too:

> [yuzu.zip](https://github.com/yuzu-emu/yuzu/files/11645992/yuzu.zip) Here's a Windows version of yuzu built against master that has the extra renderpass breaks removed. This gives me the ultrahand issue again.
> 
> ```diff
> --- a/src/video_core/renderer_vulkan/vk_texture_cache.cpp
> +++ b/src/video_core/renderer_vulkan/vk_texture_cache.cpp
> @@ -862,7 +862,7 @@ VkBuffer TextureCacheRuntime::GetTemporaryBuffer(size_t needed_size) {
>  }
> 
>  void TextureCacheRuntime::BarrierFeedbackLoop() {
> -    scheduler.RequestOutsideRenderPassOperationContext();
> +    //scheduler.RequestOutsideRenderPassOperationContext();
>  }
> 
>  void TextureCacheRuntime::ReinterpretImage(Image& dst, Image& src,
> ```
> 
> Does this resolve the device losses?

This may be of interest, as you thoroughly pointed, your system has different characteristics than the samples previously seen/reporting in here being affected by said modifications.

-------------------------

Additionally, In another situation, someone in Discord was relating a very similar circunstance earlier and given direction to try out/check the build in the discussion here. The person didn't come back to directly answer afterwards but noticed returned later and marked it with a 'heart" and a "100", so guess it worked for them, too (lol)

![Sem título](https://github.com/yuzu-emu/yuzu/assets/134863891/0d65f48d-d9ff-4065-bf87-a9e64d2f3e2c)

**Comment by liamwhite** [comment]
I played through the Great Sky Islands for 30 minutes uninterrupted on mainline 1466 with the 1080Ti on Windows 11, using driver 535.98, 16GB of RAM, and matched up my settings to yours based on the log. I assume that if the device loss was going to happen, it would have happened by now with >90% probability. I have two more things potentially different:
- I am playing over a remote connection. This might matter. Some nvidia driver bugs don't occur if you are not connected to a monitor.
- I have a 12600K in this system, which probably doesn't matter.

(Do you have a custom configuration set for yuzu in the Nvidia control panel?)

**Comment by Flexa007** [comment]
Testing without any mods, as well, approx after same 30 mins in 3661, similar crash. Took a screen of the moment rather than recording this time (0 fps in bottom):

![Sem título](https://github.com/yuzu-emu/yuzu/assets/134863891/7d21259d-b498-43c0-abd9-0828ad58cb8d)

Regarding the nvidia settings, yep, no specific profile - I only had it for testing purposes, but deleted right after. Rest is all default global settings.

Anyways, if you can't reproduce the situation, it's no worries @liamwhite. I'll understand if it's a complicated matter and you're tied and not really seeing any practical solution. I comprehend. First and foremost, I don't intend any prejudice against players who are clearly beneffiting from the change and please, be it far from me to affect those who are having their gameplay way better by it. And you gotta think about them, as well, of course. Nonetheless, the build you've provided for testing went really good, still.

**Comment by louislam125** [comment]
I am very new in playing yuzu. Using the 3661 EA build with nvidia driver version 535.98. I often encounter Vulkan Device Loss after fast travel or loading after a cutscene in FE3H.

**Comment by JJ1553** [comment]
> that cpu and gpu combo should manage solid 30, your settings are too high somewhere.

I am running 1440p, that is most likely why my fps is lower than youd imagine it to be. Other than that I believe I have all of my settings in order to that of what is reccomended. Nothing majorly out of wack to my knowledge. Ill try to get some logs and whatnot and post them here.

**Comment by louislam125** [comment]
I just tried the 3601 build and it actually fixed my Vulkan Device Loss. I wonder why 3601 is that special. Very weird.

**Comment by Flexa007** [comment]
@JJ1553 
> > that cpu and gpu combo should manage solid 30, your settings are too high somewhere.
> 
> I am running 1440p, that is most likely why my fps is lower than youd imagine it to be. Other than that I believe I have all of my settings in order to that of what is reccomended. Nothing majorly out of wack to my knowledge. Ill try to get some logs and whatnot and post them here.
> 
> > you need to have the Vulkan Validation Layer installed for the graphics debugging output to produce anything useful
> 
> Im not quite sure what is meant by having the validation layer installed. @Flexa007 would you mind sharing what you did for this? I want to make sure I have the correct log formats to share

Certainly. Basically, by installing Vulkan SDK (https://www.lunarg.com/vulkan-sdk/), then running yuzu debug with graphics debugging enabled.

Allow me to add, since you're also reporting a visible loss of performance, I believe recording in same circunstances from both the smooth and troubled versions would be most welcome, as likely would demonstrate how with same settings and everything equal in both you're having two different outcomes. 

@louislam125 
> I just tried the 3601 build and it actually fixed my Vulkan Device Loss. I wonder why 3601 is that special. Very weird.
Yep, that's exactly what we're studying. Apparently, modifications made to the render process from 3602 onwards unfortunately may have ended up affecting certain machines in unknown/unforeseen manners. 

So far, we seem to have made good progress to pinpoint and identify the cause, as seen in this thread, those rolling back to a previous version without said specific changes or trying the test build (that does not contain 3602 modifications) have been reporting it fixes/removes the issue.

**Comment by liamwhite** [comment]
I played for two hours on the 1080Ti with a 1080p screen connected and never encountered a device loss. After playing I tested the custom build and did notice that the ultrahand issue does not occur on this GPU, which is interesting since it does occur on the 1060.

**Comment by Squall-Leonhart** [comment]
theres a lot of points of divergence where things can start to misbehave, 
software wise, hardware wise.


535.98 itself has exposed weaknesses in not all together stable systems, which unfortunately is just something that can happen since testing can only infer, but not prove stability, and even when you have affirmed stability, thats only the case until you change the environment and loading behavior by changing the software running in it, hell, the Windows 7 bulldozer performance patches rendered many existing i7 overclocks unstable because even intel benefitted from changes to the scheduler there.

In my case i went from 535.50 which was an insider driver and tested stable for months (with equivalent uptime) to 535.98 and 1a off by one bit bsods started to happen.  Others reporting crashes in Dead space remastered and the few actually open (those not with the usual ignorant attitude of "if it happened after the driver update, its the drivers" mentality) have gone back and resolved those crashes with a fresh set of stability testing and tweaks to their manual memory timings, or extra voltage, I suspect the RDR2 TDR's reported and attributed to 535.98 have a similar origin.

My case just happened to be "I" decided to run 2T memory at 1T, and the imc on that system had some within spec vdroop that might have contributed to the bit errors.

I have the opportunity to do some testing with a 1080ti myself, but i doubt i'll see much deviation from Liam's findings.

performance stuff configuration out of the way, the usage of a error prone riser cable isn't off the table for introducing deviation in stability where an application changes how it does something, 
my experience with the AGP spec 20 years ago having chipset flaws on VIA 8x is always fresh in my mind when i consider how the interaction on the bus itself can screw with things,  back then you could run say a Geforce 4, FX or 6 on a KT600a and OpenGL tested perfectly, but 3DMark would corrupt and have vertex explosions because VIA screwed up Sideband addressing, which D3D actually optimised for.


Nvidia actually has some issues with PCIE ASPM on the skylake+++++ pcie controller, that manifested after 452.22, RTX/Turing gpu's are rendered quite unstable with the feature enabled while GTX 10 series would boot up and have the NVDEC reporting 100% usage until the driver was either state cycled in device manager, or a TDR occured, *Thats not to say newer PCIE controllers are that much better, the ASPM feature causes bus-drop offs with nvme's and some RTX 30 and 40 cards become progressively less stable with it on too*

**Comment by Flexa007** [comment]
> theres a lot of points of divergence where things can start to misbehave, software wise, hardware wise.
> 
> 535.98 itself has exposed weaknesses in not all together stable systems, which unfortunately is just something that can happen since testing can only infer, but not prove stability, and even when you have affirmed stability, thats only the case until you change the environment and loading behavior by changing the software running in it, hell, the Windows 7 bulldozer performance patches rendered many existing i7 overclocks unstable because even intel benefitted from changes to the scheduler there.
> 
> In my case i went from 535.50 which was an insider driver and tested stable for months (with equivalent uptime) to 535.98 and 1a off by one bit bsods started to happen. Others reporting crashes in Dead space remastered and the few actually open (those not with the usual ignorant attitude of "if it happened after the driver update, its the drivers" mentality) have gone back and resolved those crashes with a fresh set of stability testing and tweaks to their manual memory timings, or extra voltage, I suspect the RDR2 TDR's reported and attributed to 535.98 have a similar origin.
> 
> My case just happened to be "I" decided to run 2T memory at 1T, and the imc on that system had some within spec vdroop that might have contributed to the bit errors.
> 
> I have the opportunity to do some testing with a 1080ti myself, but i doubt i'll see much deviation from Liam's findings.
> 
> performance stuff configuration out of the way, the usage of a error prone riser cable isn't off the table for introducing deviation in stability where an application changes how it does something, my experience with the AGP spec 20 years ago having chipset flaws on VIA 8x is always fresh in my mind when i consider how the interaction on the bus itself can screw with things, back then you could run say a Geforce 4, FX or 6 on a KT600a and OpenGL tested perfectly, but 3DMark would corrupt and have vertex explosions because VIA screwed up Sideband addressing, which D3D actually optimised for.
> 
> Nvidia actually has some issues with PCIE ASPM on the skylake+++++ pcie controller, that manifested after 452.22, RTX/Turing gpu's are rendered quite unstable with the feature enabled while GTX 10 series would boot up and have the NVDEC reporting 100% usage until the driver was either state cycled in device manager, or a TDR occured, _Thats not to say newer PCIE controllers are that much better, the ASPM feature causes bus-drop offs with nvme's and some RTX 30 and 40 cards become progressively less stable with it on too_
> 
> Consider raising the ringbus voltage a tad, you do happen to be on one of the chips affected by the cache parity error correction bug, and i can only hope you're not on a really outdated bios with the SMT bug still present.

Thanks for the incredible input and for taking your time to explain this thoroughly, in details, @Squall-Leonhart. Much appreciated. I actually plan on changing PC sometime around next year, so hopefully it'll all go smooth, too! Plus, I'll try as you said.

Also, rest assured that the last thing I wish to be is a burden to you all, please. As you can see, I'm really trying and putting an effort to indeed attempt things and not lazy it all to the devs, as if I'm entitled or something... Hopefully, you're seeing this way, too. if not, please, pardon me. It wasn't my intention.

Time, I believe, is our most precious good/resource, so certainly I'm not trying to lose yours, by all means. And in that understanding of time as this valuable, you also have many other stuff to deal and people to attend to and new problems to think about and many other players around with many, different situations happening.... so, feel free to close this issue if you see not a viable way around here. As it's said, time flies, and I understand if you can't really keep occupying yourself with this matter. I am certainly am already very thankful for all have learned and light that was shed here.

**Comment by JJ1553** [comment]
Hey sorry for the wait on my logs. I have summer class keeping me very busy. Anyway, I ran yuzu 3601, 3602, and the test build posted here (the first one). With the graphics debugging setting on I was getting atrocious fps, <12 in lookout landing, but it would jump up in things like maps and what not. As a reminder, the main issue I noticed was significantly ~25% lower fps on EA 3602 and up, for me at 1440p it was a difference of 27-30 to 21-24 fps betwen 3601 and 3602. All settings were kept constant throughout testing. 

**Just to keep things all in one section. I am running:**
Ryzen 5 3600 (overlclocked 4.125 ghz)
Radeon 6700xt (latest driver, Overclocked to almost max allowed)
Windows 11
No HAGS (not avalible in windows)
Resize BAR enabled in BIOS
16gb of ram (overclocked to 3333mhz, tight timing)
(in addition I have disk space ram allocated in case I run out during gameplay, 20gb nvme)

**Yuzu Settings:**
![Screenshot 2023-06-13 000819](https://github.com/yuzu-emu/yuzu/assets/80869829/07f646d8-83c9-4b67-a358-9c3d0959c359)
![Screenshot 2023-06-13 000800](https://github.com/yuzu-emu/yuzu/assets/80869829/d5a45d8b-0bc2-417c-8d3f-3a45ed068ec7)
![Screenshot 2023-06-13 000733](https://github.com/yuzu-emu/yuzu/assets/80869829/69bc6d10-af41-4038-9fb7-8b98bd4865df)


**Here are the logs in question:**
[3601.txt](https://github.com/yuzu-emu/yuzu/files/11743182/3601.txt)

[3602.txt](https://github.com/yuzu-emu/yuzu/files/11743184/3602.txt)

[TEST build.txt](https://github.com/yuzu-emu/yuzu/files/11743186/TEST.build.txt)


**Additionally, here is a VKDiag output if that is useful to yall still:**
[VKDiag output.txt](https://github.com/yuzu-emu/yuzu/files/11743193/VKDiag.output.txt)

Please let me know if there is anything else I can do!

**Comment by Flexa007** [comment]
@JJ1553 yes, while running with the debugging fully enabled (including the layers) it takes a hit in performance.

Might be interesting to report your performance in test build without debugging, tho. If you noticed recovered some, then it's another evidence that adds to the current situation.

**Comment by JaJenks2** [comment]
> @JJ1553 yes, while running with the debugging fully enabled (including the layers) it takes a hit in performance.
> 
> 
> 
> Might be interesting to report your performance in test build without debugging, tho. If you noticed recovered some, then it's another evidence that adds to the current situation.
> 
> 
> 
> 

While running the test build without debugging, I was seeing the same, or even better performance as pre 3602 builds, it was just more stuttery at times.

**Comment by Flexa007** [comment]
@liamwhite much thanks for the patience and for the pull #10777. Basically, awaiting for it to be merged to return you feedback.

**Comment by Flexa007** [comment]
Said pull has been running for 14hours as of this moment, divided approx in 6h of active gameplay, followed by 5h afk and then returning to another 2h of actively playing, all continous, smooth and without a single crash nor freeze.

Therefore can say the merge (started from EA 3682) has been flawless. 

Thanks @liamwhite and @Squall-Leonhart especially for all your time, patience and dedication to share your work and knowledge during this course. Thanks also to the others who reported and contributed.

Plus, as approx nearly a day since merge with no reports of drawbacks and since the others also haven't indicated/comunicated further ramifications, thus closing this one.

_(For further reference/documentation in case anyone reads this in future, a fix was implemented in EA 3682 in the form of option "barrier feedback loops", under adv. graphics tab. Remember if you're having same trouble and researching)._

**Comment by LOuroboros** [comment]
I hadn't had time to test the latest EA builds.

I did check out the EA 3689 just now and got a Device Loss error not even 15 minutes after resuming my savefile of Persona 5 Royal.

To double check, am I supposed to keep "Barrier feedback loops" enabled, or disabled? Which one gives me a behavior that is the closest to whatever [the first build Liam dropped in this PR](https://github.com/yuzu-emu/yuzu/issues/10600#issuecomment-1575611653)?

For reference, I have the option disabled.

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11778103/yuzu_log.txt)

**Comment by louislam125** [comment]
> I hadn't had time to test the latest EA builds.
> 
> I did check out the EA 3689 just now and got a Device Loss error not even 15 minutes after resuming my savefile of Persona 5 Royal.
> 
> To double check, am I supposed to keep "Barrier feedback loops" enabled, or disabled? Which one gives me a behavior that is the closest to whatever [the first build Liam dropped in this PR](https://github.com/yuzu-emu/yuzu/issues/10600#issuecomment-1575611653)?
> 
> For reference, I have the option disabled.
> 
> [yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11778103/yuzu_log.txt)
> 
> **EDIT**: Oh, one more question; what is the best driver version for a GPU such as mine? Should I stay in 512.95, or try my luck updating to the latest version?

I just encounter it in 3685 too with "Barrier feedback loops" enabled too. But i am in 535.98. I am still hesitate if I should install the latest driver.

**Comment by Flexa007** [comment]
@LOuroboros 
> I hadn't had time to test the latest EA builds.
> 
> I did check out the EA 3689 just now and got a Device Loss error not even 15 minutes after resuming my savefile of Persona 5 Royal.
> 
> To double check, am I supposed to keep "Barrier feedback loops" enabled, or disabled? Which one gives me a behavior that is the closest to whatever [the first build Liam dropped in this PR](https://github.com/yuzu-emu/yuzu/issues/10600#issuecomment-1575611653)?
> 
> For reference, I have the option disabled.
> 
> [yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11778103/yuzu_log.txt)
> 
> **EDIT**: Oh, one more question; what is the best driver version for a GPU such as mine? Should I stay in 512.95, or try my luck updating to the latest version?

Apparently, another circunstance is triggering it now @LOuroboros. When you did post your previous log, it did have the exact same device loss report as mine, that seems to be the reason as to why the test build thus fixed it for both of us. However, your current log is now reporting a different device loss. Since you were able to play perfectly with test build, this possibly might mean another change that happened in between is now conflicting. 

You might want to try all the way back to 3682 (barrier option implementation), and if you find the build that started causing this device loss (and hopefully you do) it'll be "easier" to comprehend, as you will be able to know exactly which build had a transition that modified something that didn't sit well with your rig. For example, knowing last time it worked was 36xx but something changed in the immediate one 36xy that caused issues. 

If you manage, you should put together your logs and evidence and open up a new issue in here.

Unfortunately, if still crashing, it'll be trickier to identify since it'll likely be something in between test build date and 3680 (last build before the workaround) when you still also had barrier conflict, as well.

-------------------------
> I just encounter it in 3685 too with "Barrier feedback loops" enabled too. But i am in 535.98. I am still hesitate if I should install the latest driver.

@louislam125 You gotta run with it unchecked. Log file would be helpful, might add. 535.98 should be fine, too. The test build basically removes the changes in between EAs 3601 and 3602, and if it's still happening, might be exact same situation as above. A new source of conflict.

**Comment by Squall-Leonhart** [comment]
you want to **disable** barrier feedback loops to achieve the 3601 behavior.

**Comment by louislam125** [comment]
@Flexa007 Here is the log with Barrier feedback loops disabled. Game's running good in 3601 but later builds ramdomly cause device loss which fe3h freezes in loading screen most of the time.
[yuzu_log.txt.old.txt](https://github.com/yuzu-emu/yuzu/files/11778302/yuzu_log.txt.old.txt)

**Comment by Flexa007** [comment]
> @Flexa007 Here is the log with Barrier feedback loops disabled. Game's running good in 3601 but later builds ramdomly cause device loss which fe3h freezes in loading screen most of the time. [yuzu_log.txt.old.txt](https://github.com/yuzu-emu/yuzu/files/11778302/yuzu_log.txt.old.txt)

Given it doesn't work since 3601's transition, then it indeed cements you had a barrier conflict, previously. With the current skip implementation, that source of incomptability is gone. This seems the case, as your log is now showing the same device loss report as @LOuroboros' lastest one. This makes it likely another change, possibly the same, that happened in between ended up affecting you both.

**Comment by louislam125** [comment]
> @LOuroboros
> 
> > I hadn't had time to test the latest EA builds.
> > I did check out the EA 3689 just now and got a Device Loss error not even 15 minutes after resuming my savefile of Persona 5 Royal.
> > To double check, am I supposed to keep "Barrier feedback loops" enabled, or disabled? Which one gives me a behavior that is the closest to whatever [the first build Liam dropped in this PR](https://github.com/yuzu-emu/yuzu/issues/10600#issuecomment-1575611653)?
> > For reference, I have the option disabled.
> > [yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11778103/yuzu_log.txt)
> > **EDIT**: Oh, one more question; what is the best driver version for a GPU such as mine? Should I stay in 512.95, or try my luck updating to the latest version?
> 
> Apparently, another circunstance is triggering it now @LOuroboros. When you did post your previous log, it did have the exact same device loss report as mine, that seems to be the reason as to why the test build thus fixed it flawlessly for both of us. However, your current log is now reporting a different device loss. Since you were able to play perfectly with test build, this possibly might mean another change that happened in between is now conflicting.
> 
> You might want to try all the way back to 3682 (skip barrier option implementation), and if you find the build that started causing this device loss (and hopefully you do) it'll be "easier" to comprehend, as you will be able to know exactly which build had a transition that modified something that didn't sit well with your rig. For example, knowing last time it worked was 36xx but something changed in the immediate one 36xy that caused issues.
> 
> If you manage, you should put together your logs and evidence and open up a new issue in here.
> 
> Unfortunately, if still crashing, it'll be trickier to identify since it'll likely be something in between test build date and 3680 (last build before the workaround) when you still also had barrier conflict, as well.
> 
> > I just encounter it in 3685 too with "Barrier feedback loops" enabled too. But i am in 535.98. I am still hesitate if I should install the latest driver.
> 
> @louislam125 You gotta run with it unchecked. Log file would be helpful, might add. 535.98 should be fine, too. The test build basically removes the changes in between EAs 3601 and 3602, and if it's still happening, might be exact same situation as above. A new source of conflict.

So I just tried 3682 and the same thing happened.

**Comment by Flexa007** [comment]
> > @LOuroboros
> > > I hadn't had time to test the latest EA builds.
> > > I did check out the EA 3689 just now and got a Device Loss error not even 15 minutes after resuming my savefile of Persona 5 Royal.
> > > To double check, am I supposed to keep "Barrier feedback loops" enabled, or disabled? Which one gives me a behavior that is the closest to whatever [the first build Liam dropped in this PR](https://github.com/yuzu-emu/yuzu/issues/10600#issuecomment-1575611653)?
> > > For reference, I have the option disabled.
> > > [yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11778103/yuzu_log.txt)
> > > **EDIT**: Oh, one more question; what is the best driver version for a GPU such as mine? Should I stay in 512.95, or try my luck updating to the latest version?
> > 
> > 
> > Apparently, another circunstance is triggering it now @LOuroboros. When you did post your previous log, it did have the exact same device loss report as mine, that seems to be the reason as to why the test build thus fixed it flawlessly for both of us. However, your current log is now reporting a different device loss. Since you were able to play perfectly with test build, this possibly might mean another change that happened in between is now conflicting.
> > You might want to try all the way back to 3682 (skip barrier option implementation), and if you find the build that started causing this device loss (and hopefully you do) it'll be "easier" to comprehend, as you will be able to know exactly which build had a transition that modified something that didn't sit well with your rig. For example, knowing last time it worked was 36xx but something changed in the immediate one 36xy that caused issues.
> > If you manage, you should put together your logs and evidence and open up a new issue in here.
> > Unfortunately, if still crashing, it'll be trickier to identify since it'll likely be something in between test build date and 3680 (last build before the workaround) when you still also had barrier conflict, as well.
> > > I just encounter it in 3685 too with "Barrier feedback loops" enabled too. But i am in 535.98. I am still hesitate if I should install the latest driver.
> > 
> > 
> > @louislam125 You gotta run with it unchecked. Log file would be helpful, might add. 535.98 should be fine, too. The test build basically removes the changes in between EAs 3601 and 3602, and if it's still happening, might be exact same situation as above. A new source of conflict.
> 
> So I just tried 3682 and the same thing happened.

Then likely it is something in between 3603 (first post 3602 barrier) and 3680 (last pre skip barrier 3682).

The barrier was, up to some versions ago, your sole source, this is why 3601 works. And since such version works perfectly with no other troubles, this also means another change that thus happened after, in that window mentioned above (3603-3680), started causing another conflict. Given your and @LOuroboros new logs are reporting same device loss, it is also likely it might be the same culprit for both. 

It is just very tricky to identify because running within such window you'll also crash because of the barrier, as well. In this case, having to be very honest, gotta raise aware to bear in mind this will likely have no quick solution as this seems certainly a situation that will demand a lot of patience and time to investigate (and that being just to identify, as said).

If you have building/compiling knowledge, might have to redo previous builds, but editing the barrier out, till you find which one starts crashing again, and with it finally pinpointed, inspect the release notes for said build and you'll possibly identify the change that is the source.

Given the original problem, the barrier, has been solved and this has a different source now, it is why it is better to proceed in another issue as to not mix things.

**Comment by Squall-Leonhart** [comment]
> **EDIT**: Oh, one more question; what is the best driver version for a GPU such as mine? Should I stay in 512.95, or try my luck updating to the latest version?

All pascal cards were unstable after 472, till 526, so its best to be on the latest driver.

**Comment by louislam125** [comment]
> > > @LOuroboros
> > > > I hadn't had time to test the latest EA builds.
> > > > I did check out the EA 3689 just now and got a Device Loss error not even 15 minutes after resuming my savefile of Persona 5 Royal.
> > > > To double check, am I supposed to keep "Barrier feedback loops" enabled, or disabled? Which one gives me a behavior that is the closest to whatever [the first build Liam dropped in this PR](https://github.com/yuzu-emu/yuzu/issues/10600#issuecomment-1575611653)?
> > > > For reference, I have the option disabled.
> > > > [yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11778103/yuzu_log.txt)
> > > > **EDIT**: Oh, one more question; what is the best driver version for a GPU such as mine? Should I stay in 512.95, or try my luck updating to the latest version?
> > > 
> > > 
> > > Apparently, another circunstance is triggering it now @LOuroboros. When you did post your previous log, it did have the exact same device loss report as mine, that seems to be the reason as to why the test build thus fixed it flawlessly for both of us. However, your current log is now reporting a different device loss. Since you were able to play perfectly with test build, this possibly might mean another change that happened in between is now conflicting.
> > > You might want to try all the way back to 3682 (skip barrier option implementation), and if you find the build that started causing this device loss (and hopefully you do) it'll be "easier" to comprehend, as you will be able to know exactly which build had a transition that modified something that didn't sit well with your rig. For example, knowing last time it worked was 36xx but something changed in the immediate one 36xy that caused issues.
> > > If you manage, you should put together your logs and evidence and open up a new issue in here.
> > > Unfortunately, if still crashing, it'll be trickier to identify since it'll likely be something in between test build date and 3680 (last build before the workaround) when you still also had barrier conflict, as well.
> > > > I just encounter it in 3685 too with "Barrier feedback loops" enabled too. But i am in 535.98. I am still hesitate if I should install the latest driver.
> > > 
> > > 
> > > @louislam125 You gotta run with it unchecked. Log file would be helpful, might add. 535.98 should be fine, too. The test build basically removes the changes in between EAs 3601 and 3602, and if it's still happening, might be exact same situation as above. A new source of conflict.
> > 
> > 
> > So I just tried 3682 and the same thing happened.
> 
> Then likely it is something in between 3603 (first post 3602 barrier) and 3680 (last pre skip barrier 3682).
> 
> Given what you described/showed, the barrier was, up to some versions ago, your sole source, this is why 3601 works. And since such version works perfectly with no other troubles, this also means another change that thus happened after, in that window mentioned above (3603-3680), started causing another conflict. Given your and @LOuroboros new logs are reporting same device loss, it is also likely it might be the same culprit for both.
> 
> It is just very tricky to identify because running within such window you'll also crash because of the barrier, as well. In this case, having to be very honest, gotta raise aware to bear in mind this will likely have no quick solution as this seems certainly a situation that will demand a lot of patience and take no small amount of time to investigate (and that being just to identify, as said).
> 
> If you have building/compiling knowledge, might have to redo previous builds, obviously within the window above, but editing the barrier out (so you don't crash because of it), till you find which one starts crashing again, and with it finally pinpointed, inspect the release notes for said build and you'll possibly identify the change that is the source.
> 
> Given the original problem, the barrier, has been solved and this has a different source now, it is why it is better to proceed in another issue as to not mix things.

Thanks for the long reply. I couldn't figure it out myself because update occurs too frequently. You never know what update fix things or break things.

**Comment by Squall-Leonhart** [comment]
It appears that R535+ drivers have rendered attachment feedback loops unnecessary on at least Pascal hardware, 

The images i previously took and posted here are a glitch that happens even on hardware when using ultrahand underground or across uneven surfaces (the green overlay is projected against wall surfaces and can even be seen in towns)

And I found a crasher with the feedback loops enabled, in Sonic Origins where the game could crash within moments or after finishing a level.
